--- target/linux/netfilter.mk	2006-12-19 09:52:59.000000000 -0500
+++ target/linux/netfilter.mk	2008-04-29 00:36:38.669377980 -0400
@@ -80,6 +80,8 @@
 IPT_NAT_EXTRA-m := 
 IPT_NAT_EXTRA-$(CONFIG_IP_NF_AMANDA) += ip_conntrack_amanda
 IPT_NAT_EXTRA-$(CONFIG_IP_NF_NAT_SNMP_BASIC) += ip_nat_snmp_basic
+IPT_NAT_EXTRA-$(CONFIG_IP_NF_NAT_SL) += ip_nat_sl
+IPT_NAT_EXTRA-$(CONFIG_IP_NF_NAT_SLPORT) += ip_nat_slport
 
 IPT_QUEUE-m :=
 IPT_QUEUE-$(CONFIG_IP_NF_QUEUE) += ip_queue
