Index: package/kernel/modules/netfilter.mk
===================================================================
--- package/kernel/modules/netfilter.mk	(revision 11949)
+++ package/kernel/modules/netfilter.mk	(working copy)
@@ -107,7 +107,23 @@
 
 $(eval $(call KernelPackage,ipt-nat))
 
+define KernelPackage/ipt-natsl
+  SUBMENU:=$(NF_MENU)
+  TITLE:=Silver Lining Conntrack and NAT helpers
+  KCONFIG:=$(KCONFIG_IPT_NATSL)
+  FILES:=$(foreach mod,$(IPT_NATSL-m),$(LINUX_DIR)/net/$(mod).$(LINUX_KMOD_SUFFIX))
+  AUTOLOAD:=$(call AutoLoad,40,$(notdir $(IPT_NATSL-m)))
+endef
 
+define KernelPackage/ipt-natsl/description
+ Silver Lining Netfilter (IPv4) Conntrack and NAT helpers
+ Includes:
+ - nf_conntrack_sl
+ - nf_nat_sl
+endef
+
+$(eval $(call KernelPackage,ipt-natsl))
+
 define KernelPackage/ipt-nathelper
   SUBMENU:=$(NF_MENU)
   TITLE:=Default Conntrack and NAT helpers
